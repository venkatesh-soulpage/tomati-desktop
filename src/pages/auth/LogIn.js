import React from "react";
// Redux
import { connect } from "react-redux";
import {
  userLogin,
  handleLoginError,
  receiveUserData,
  clearLoginError,
} from "_actions/auth";
// Router
import { withRouter, Link } from "react-router-dom";
// Bootstrap Components
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
// custom components
import PasswordTextField from "components/PasswordTextField";
import AlertMessage from "components/AlertMessage";
import Footer from "components/Footer";

function LogIn(props) {
  // Form states
  const [values, setValues] = React.useState({
    email: "",
    password: "",
    userLoginResponse: {},
  });

  React.useEffect(function () {
    window.scroll(0, 0);
    props.dispatch(receiveUserData({}));
    props.dispatch(handleLoginError(null));
  }, []);

  // For handling changes in the inputs
  const handleChange = (name) => (event) => {
    const value = event.target.value;
    setValues((values) => ({ ...values, [name]: value }));
  };

  // Handling the login data and sending it to the service.
  function handleLoginData(event) {
    event.preventDefault();
    // Creating post Data
    var postData = {
      email: values.email,
      password: values.password,
    };
    props
      .dispatch(userLogin(postData))
      .then((userData) => {
        if (userData.email_verified_at === null) {
          props.history.push("/verify-email");
        } else if (userData.sms_verified_at === null) {
          props.history.push("/verify-phone");
        } else if (sessionStorage.getItem("token")) {
          props.history.push("/dashboard");
        }
      })
      .catch((error) => {});
  }

  function handleAlertDismiss() {
    props.dispatch(clearLoginError());
  }

  return (
    <div className="min-height mt-5">
      <div className="container-fluid h-100">
        <div className="row h-100 justify-content-center">
          <div className="col-md-4 w-100 align-self-center">
            <Card className="rounded shadow">
              <Form
                onSubmit={handleLoginData}
                autoComplete="off"
                className="p-5"
              >
                <AlertMessage
                  variant="danger"
                  error={props.auth.loginError}
                  onDismiss={handleAlertDismiss}
                />
                <Form.Group>
                  <Form.Control
                    type="email"
                    name="email"
                    id="email"
                    value={values.email}
                    onChange={handleChange("email")}
                    placeholder="Enter email"
                  />
                </Form.Group>
                <Form.Group>
                  <PasswordTextField
                    name="password"
                    value={values.password}
                    onChange={handleChange("password")}
                    placeholder="Password"
                  />
                </Form.Group>
                <Form.Group className="d-flex justify-content-between">
                  <Link to="/forgot-password" style={{ color: "#E0475B" }}>
                    Forgot Password?
                  </Link>
                  <Button type="submit" style={{ borderRadius: "20px" }}>
                    Login
                  </Button>
                </Form.Group>
              </Form>
            </Card>
          </div>
        </div>
      </div>
    </div>
  );
}

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default withRouter(connect(mapStateToProps)(LogIn));
