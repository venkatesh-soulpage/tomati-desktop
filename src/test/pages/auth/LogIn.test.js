import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import LogIn from "pages/auth/LogIn";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("LogIn page", () => {
  it("before entering values", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <LogIn />
        </Router>
      </Provider>,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("after entering values", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <LogIn />
        </Router>
      </Provider>,
      {}
    );
    let inputNode_email = screen.getByPlaceholderText("Enter email");
    let inputNode_password = screen.getByPlaceholderText("Password");
    fireEvent.change(inputNode_email, {
      target: {
        value: "tomati@gmail.com",
      },
    });
    fireEvent.change(inputNode_password, {
      target: {
        value: "Soul@123",
      },
    });
    expect(inputNode_email.value).toBe("tomati@gmail.com");
    expect(inputNode_password.value).toBe("Soul@123");

    expect(asFragment()).toMatchSnapshot();
  });
});
