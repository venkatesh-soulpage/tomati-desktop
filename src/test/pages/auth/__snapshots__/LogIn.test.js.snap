// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn page after entering values 1`] = `
<DocumentFragment>
  <div
    class="container-fluid min-height d-flex align-items-center justify-content-center"
  >
    <div
      class="col-md-4 w-100 p-0 "
    >
      <h2
        class="text-dark text-center mb-4"
      >
        Login
      </h2>
      <div
        class="p-md-5 pb-md-3 p-2 login-card card-border card"
      >
        <form
          autocomplete="off"
          class=""
        >
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="bg-white border-right-0 input-group-text"
                >
                  <img
                    alt="mail"
                    class="img-fluid"
                    height="20"
                    src="Mail.svg"
                    width="20"
                  />
                </span>
              </div>
              <input
                class="border-left-0 form-control"
                id="email"
                name="email"
                placeholder="Enter email"
                required=""
                type="email"
                value="tomati@gmail.com"
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="bg-white border-right-0 input-group-text"
                >
                  <img
                    alt="lock"
                    class="img-fluid"
                    height="20"
                    src="Lock.svg"
                    width="20"
                  />
                </span>
              </div>
              <input
                class="border-left-0 border-right-none form-control"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value="Soul@123"
              />
              <div
                class="input-group-append"
              >
                <div
                  class="show-button"
                >
                  <small>
                    Show
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between align-items-center mt-4 form-group"
          >
            <a
              class=" red-link"
              href="/forgot-password"
            >
              Forgot Password?
            </a>
            <button
              class="rounded-pill btn-danger px-4 btn btn-primary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LogIn page before entering values 1`] = `
<DocumentFragment>
  <div
    class="container-fluid min-height d-flex align-items-center justify-content-center"
  >
    <div
      class="col-md-4 w-100 p-0 "
    >
      <h2
        class="text-dark text-center mb-4"
      >
        Login
      </h2>
      <div
        class="p-md-5 pb-md-3 p-2 login-card card-border card"
      >
        <form
          autocomplete="off"
          class=""
        >
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="bg-white border-right-0 input-group-text"
                >
                  <img
                    alt="mail"
                    class="img-fluid"
                    height="20"
                    src="Mail.svg"
                    width="20"
                  />
                </span>
              </div>
              <input
                class="border-left-0 form-control"
                id="email"
                name="email"
                placeholder="Enter email"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="bg-white border-right-0 input-group-text"
                >
                  <img
                    alt="lock"
                    class="img-fluid"
                    height="20"
                    src="Lock.svg"
                    width="20"
                  />
                </span>
              </div>
              <input
                class="border-left-0 border-right-none form-control"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <div
                class="input-group-append"
              >
                <div
                  class="show-button"
                >
                  <small>
                    Show
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between align-items-center mt-4 form-group"
          >
            <a
              class=" red-link"
              href="/forgot-password"
            >
              Forgot Password?
            </a>
            <button
              class="rounded-pill btn-danger px-4 btn btn-primary"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
